api/napi/index.ts

```diff
@@ -144,4 +145,101 @@ export class ArrayModel<T> implements Model<T> {
     entries(): IterableIterator<[number, T]> {
         return this.a.entries()
     }
-}
\ No newline at end of file
+}
+
+export interface ComponentHandle {
+    run();
+    show();
+    hide();
+    get window(): Window;
+}
+
+class Component implements ComponentHandle {
+    private instance: ComponentInstance;
+
+    /**
+    * @hidden
+    */
+    constructor(instance: ComponentInstance) {
+        this.instance = instance;
+    }
+
+    run() {
+        this.instance.run();
+    }
+
+    show() {
+        this.instance.window().show();
+    }
+
+    hide() {
+        this.instance.window().hide();
+    }
+
+    get window(): Window {
+        return this.instance.window();
+    }
+}
+
+/**
+ * @hidden
+ */
+interface Callback {
+    (): any;
+    setHandler(cb: any): void;
+}
+
+export function loadFile(path: string) : Object {
+    let compiler = new ComponentCompiler;
+    let definition = compiler.buildFromPath(path);
+
+
+    let diagnostics = compiler.diagnostics;
+
+    if (diagnostics.length > 0) {
+        diagnostics.forEach((d) => {
+            if (d.level == DiagnosticLevel.Warning) {
+                console.log(d.message);
+            } else {
+                throw Error(d.message);
```

Comment:
That doesn't seem right. Instead of the user getting access to all error diagnostics, this just aborts on the first error.

Please add a unit test for this function, that tests the different code paths of success and errors.

It's a shame that the type system doesn't allow for specifying the signature of the exception, but I believe it can be done via [documentation](https://tsdoc.org/pages/tags/throws/).